renderer.php

```diff
@@ -176,6 +177,38 @@ public function render_envbar($match, $fixed = true, $envs = array()) {
             $showtext .= '<nobr> ' . $config->stringseparator . ' ' . $editlink . '</nobr>';
         }
 
+        $debuggingonstr = get_string('debuggingon', 'local_envbar');
+        $debuggingoffstr = get_string('debuggingoff', 'local_envbar');
+        $debuggingdefinedstr = get_string('debuggingdefinedinconfig', 'local_envbar');
+        $debugging = envbarlib::get_debugging_status_string($CFG->debug);
+        // Show debugging links for admins.
+        if ($canedit && $config->showdebugging) {
+            if ($debugging === $debuggingonstr) {
+                $debugtogglestr = get_string('debugtogglelinkoff', 'local_envbar');
+            } else {
+                $debugtogglestr = get_string('debugtogglelinkon', 'local_envbar');
+            }
+            // Get the url of the current page.
+            $currentlink = $CFG->wwwroot . $ME;
+            $debugtogglelink = html_writer::link(
+                new moodle_url('/local/envbar/toggle_debugging.php?redirect='.$currentlink),
+                $debugtogglestr
+            );
+            // Check if debug level and debug display is set on config.php.
+            if (!isset($CFG->config_php_settings['debug']) && !isset($CFG->config_php_settings['debugdisplay'])) {
+                $showtext .= '<nobr> ' . $config->stringseparator . ' ' . $debugging. ' ' . $debugtogglelink . '</nobr>';
+            } else {
+                // Remove link to toggle debugging.
+                $showtext .= '<nobr> ' . $config->stringseparator . ' ' . $debugging;
+                $showtext .= ' ' . $debuggingdefinedstr . '</nobr>';
+            }
+        } else {
+            if ($config->showdebugging) {
```

Comment:
you can probably move this  if ($config->showdebugging) level up. If it's not set, then there is no point to do anything else.

like 

` 

if ($config->showdebugging) {
         $showtext .= $this->get_debug_text($canedit);
} 

`